beacon-chain/blockchain/process_block.go

```diff
@@ -75,6 +75,23 @@ func (s *Service) postBlockProcess(cfg *postBlockProcessConfig) error {
 	defer reportProcessingTime(startTime)
 	defer reportAttestationInclusion(cfg.roblock.Block())
 
+	// Check for equivocation before inserting into fork choice
+	slashing, slashingErr := s.detectEquivocatingBlock(cfg.ctx, cfg.roblock)
+	if slashingErr != nil {
+		return errors.Wrap(slashingErr, "could not detect equivocating block")
+	}
+
+	if slashing != nil {
```

Comment:
Better if the broadcasting logic is also handled by the helper so that it's encapsulated, the helper should call a broadcasting routine instead of returnning the slashing.

Reply:
Addressed this in the next commit